spacemouse-keys/hidInterface.cpp

```diff
@@ -4,25 +4,167 @@
 // Include inbuilt Arduino HID library by NicoHood: https://github.com/NicoHood/HID
 #include "HID.h"
 
+// Send a HID report every 8 ms (found on an original SpaceNavigator)
+#define HIDUPDATERATE_MS 8
+
+// little state machine to track, which report to send next
+enum States
+{
+  ST_START,     // start to check if something is to be sent
+  ST_SENDTRANS, // send translations
+  ST_SENDROT,   // send rotations
+  ST_SENDKEYS   // send keys
+};
+
+States nextState = ST_START; // init state machine with start state
+
 #if (NUMKEYS > 0)
-// Array with the bitnumbers, which should be assign keys to buttons
+// Array with the bitnumbers, which should assign keys to buttons
 uint8_t bitNumber[NUMHIDKEYS] = BUTTONLIST;
+
+void prepareKeyBytes(uint8_t *keys, uint8_t *keyData, int debug);
 #endif
 
-// Function to send translation and rotation data to the 3DConnexion software using the HID protocol outlined earlier. Two sets of data are sent: translation and then rotation.
+uint8_t countTransZeros = 0; // count how many times, the zero data has been sent
+uint8_t countRotZeros = 0;
+
+// Function to send translation and rotation data to the 3DConnexion software using the HID protocol
+// Three sets of data are sent: translation, rotation and key events
 // For each, a 16bit integer is split into two using bit shifting. The first is mangitude and the second is direction.
-void send_command(int16_t rx, int16_t ry, int16_t rz, int16_t x, int16_t y, int16_t z, uint8_t *keys, int debug)
+// With regard to the _hidReportDescriptor: the values should be within +/- 350.
+// returns true, if new data was sent
+// A HID report may be sent every HIDUPDATERATE_MS, which should be 8 ms
+bool send_command(int16_t rx, int16_t ry, int16_t rz, int16_t x, int16_t y, int16_t z, uint8_t *keys, int debug)
 {
-  uint8_t trans[6] = {(byte)(x & 0xFF), (byte)(x >> 8), (byte)(y & 0xFF), (byte)(y >> 8), (byte)(z & 0xFF), (byte)(z >> 8)};
-  HID().SendReport(1, trans, 6);
-  uint8_t rot[6] = {(byte)(rx & 0xFF), (byte)(rx >> 8), (byte)(ry & 0xFF), (byte)(ry >> 8), (byte)(rz & 0xFF), (byte)(rz >> 8)};
-  HID().SendReport(2, rot, 6);
+  static unsigned long now = 0; // time from millis()
+  now = millis();
```

Comment:
those 2 lines can be merged

Reply:
They can't be merged, as long as the variable is static. 
But there is no need to be static.
-> merged them in the next commit.